packages/block-library/src/heading/style.scss

```diff
@@ -9,8 +9,35 @@
 		color: var(--wp--color--text);
 		line-height: var(--wp--typography--line-height);
 	}
+
+	// These follow a Major Third type scale
+	// https://type-scale.com/?size=16&scale=1.250&text=A%20Visual%20Type%20Scale&font=Noto%20Serif&fontweight=600
+	h1 {
+		font-size: var(--wp--typography--font-size, 2.44em);
```

Comment:
The problem with this is that if this variable is defined in a the theme.json, all headings will have the same font size. Not sure what solution we should have here.

Reply:
How would you define heading sizes in theme.json files?

Reply:
> all headings will have the same font size

That's what I noticed as well 🤔 .
Do this values need a `calc()` to compute the font-scaling?

Reply:
I believe the styles here are correct, but we should try to take this as an opportunity to work on the "CSS generation plugins"

by this I mean when a theme does

```
typography: { fontSize: '12px' }
```

The global styles CSS generation mechanism would also run a heading plugin that automatically outputs

```
h1 {
--wp--typography--font-size: calc( 12 * h1Scale )
}

h2 {
--wp--typography--font-size: calc( 12 * h2Scale )
}
```

Reply:
That would be fantastic!

Reply:
We should also have a way to override these "scales" on theme.json file somehow.

Reply:
Noting here that, as it currently stands, `--wp--typography--font-size` is only present at the block-level, there's no such a variable at the global level.

This is something we need to consolidate yet ([see thread](https://github.com/WordPress/gutenberg/pull/20530#discussion_r400840203)) and there's even feedback that proposes to back down the base/scale approach to something more tied to the headings themselves. Because I don't know what will be the end result there, I don't have a concrete answer for this yet. I do have some ideas such as using something along the lines of the [block selectors PR](https://github.com/WordPress/gutenberg/pull/20290), or/and converting the scale/base to the font-size as Riad suggests above (see also [related example](https://github.com/WordPress/gutenberg/pull/20530/files#diff-f2f5efb13e0e16e61312aa9b0c028ae1R27)).

Reply:
My point here is that I think it's safe to land this PR as it is (it works like in paragraph) and, when the global level debate settled, we need to make the changes to this and paragraph.

Reply:
We're changing gears as per https://github.com/WordPress/gutenberg/pull/21428 so I'm removing this from the PR.